aten/src/ATen/native/cpu/zmath.h

```diff
@@ -93,6 +93,16 @@ constexpr double real_impl <c10::complex<double>, double> (c10::complex<double>
   return z.real();
 }
 
+template<>
+inline c10::complex<Half> real_impl <c10::complex<Half>> (c10::complex<Half> z) {
+  return c10::complex<Half>(z.real(), 0.0);
+}
+
+template<>
+inline Half real_impl <c10::complex<Half>, Half> (c10::complex<Half> z) {
```

Comment:
why aren't these `contexpr` like the `real_impl` and `imag_impl` functions for complex float and double below?